niftilib/nifti1_io.c

```diff
@@ -4405,15 +4405,15 @@ static int nifti_read_extensions( nifti_image *nim, znzFile fp, int remain )
       return -1;
    }
 
-   znz_off_t posn = znztell(fp);
+   posn = (int)znztell(fp);
```

Comment:
`znztell` returns `znz_off_t`.  The cast you've added could cause information loss.

You mentioned "znz_off_t varies, so try not to print" which I guess is why you made this change...

But what's wrong with using `%lld` for `znz_off_t`?  If `znz_off_t` is smaller than `long long` it'll still get printed correctly.

With this change, if `znztell` returns a huge number, your error message won't even print the actual thing `znztell` returned.

Reply:
posn is required to be 358, anything else is a fatal error.  And %lld will produce compiler warnings on many systems (since znz_off_t varies).  Using int64_t would be okay, but seems like overkill.

Reply:
Oh wait, it is not a fatal error, just a warning (since we don't care soooo much about extensions, I guess)...

Reply:
> And %lld will produce compiler warnings on many systems (since znz_off_t varies)

Ah true, -Wformat will warn about that. Then the most correct thing is to:

1. not do the truncating cast you added
2. cast the %lld param to `long long` to match `lld`

```c
znz_off_t posn = znztell(fp);

fprintf(stderr,"-d nre: posn = %lld, offset = %d, type = %d, remain = %d\n",
    (long long)posn, nim->iname_offset, nim->nifti_type, remain);
```

Reply:
Okay, will do.  I sill worry about getting warnings, we will see.

Reply:
`-Wformat` won't warn then, I tried.

Reply:
These things can vary across systems, as the sizes vary.  "long" types do not seem to be so reliable.

Reply:
Their size varies, but "%lld" means "long long" regardless of what the size of long long is.